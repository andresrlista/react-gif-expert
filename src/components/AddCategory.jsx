import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
  const [categoryValue, setCategoryValue] = useState('');

  const onInputChangeCategory = ({ target }) => {
    setCategoryValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (categoryValue.trim().length <= 1) return;
    onNewCategory(categoryValue.trim());
    setCategoryValue('');
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        name='categpria'
        id='categoria'
        type='text'
        value={categoryValue}
        placeholder='Burscar gifs...'
        onChange={onInputChangeCategory}
      ></input>
    </form>
  );
};
