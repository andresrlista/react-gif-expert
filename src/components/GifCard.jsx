import PropTypes from 'prop-types';

export const GifCard = ({ id, title, url }) => {
  return (
    <div key={id} className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
