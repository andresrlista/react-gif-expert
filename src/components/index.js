/* Exporting all the components from the components folder. */
export * from './AddCategory';
export * from './GifCard';
export * from './GifGrid';
