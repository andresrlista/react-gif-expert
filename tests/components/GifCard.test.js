import { getByRole, render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

describe('Pruebas en el componente GifCard', () => {
  const title = 'test';
  const url = 'https://url.test.com/';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('El title y url deben coincidir', () => {
    render(<GifCard title={title} url={url} />);
    const { src, alt } = screen.getByRole('img');
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });
});
