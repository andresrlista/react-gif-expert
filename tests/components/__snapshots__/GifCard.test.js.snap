// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifCard Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://url.test.com/"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
